llvm/lib/Target/RISCV/RISCVTargetMachine.cpp

```diff
@@ -80,6 +80,11 @@ static cl::opt<bool> EnableRISCVDeadRegisterElimination(
              " them with stores to x0"),
     cl::init(true));
 
+// TODO: This should be controlled by -mtls-dialect=<option>
+cl::opt<bool> EnableRISCVTLSDESC("riscv-enable-tlsdesc",
```

Comment:
We should add an option to `codegen::InitTargetOptionsFromCodeGenFlags`, then reference it using something like `DAG.getTarget().useTLSDESC()`. aarch64 ELF only implements TLSDESC for dynamic TLS models, it is fine for it to ignore `useTLSDESC`.

Reply:
That's a good suggestion. I was thinking of doing something like that as a follow up for the Clang driver patches, but I agree, its probably better to do it in this patch.